pkg/jobs/adopt.go

```diff
@@ -423,18 +424,53 @@ func (r *Registry) runJob(
 	return err
 }
 
+const clearClaimQuery = `
+   UPDATE system.jobs
+      SET claim_session_id = NULL, claim_instance_id = NULL
+    WHERE id = $1
+      AND claim_session_id = $2
+      AND claim_instance_id = $3
+      AND status NOT IN ('` + string(StatusPauseRequested) + `', '` + string(StatusCancelRequested) + `')`
+
+// maybeClearLease clears the claim on the given job, provided that
+// the current lease matches our liveness Session.
+func (r *Registry) maybeClearLease(job *Job, jobErr error) {
+	if jobErr == nil {
+		return
+	}
+
+	// We use the serverCtx here rather than the context from the
+	// caller since the caller's context may have been canceled.
+	r.withSession(r.serverCtx, func(ctx context.Context, s sqlliveness.Session) {
+		err := r.db.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
+			if err := txn.SetUserPriority(roachpb.MinUserPriority); err != nil {
```

Comment:
Why this?

Reply:
I was mostly following what we do in other lifecyle queries in this package (claimJobs and servePauseAndCancelRequests).  However, those happen constantly on all nodes so are more likely to cause repeated contention whereas one hopes this is relatively infrequently called, so perhaps we can do without it.

Reply:
Yeah, the other point queries in this package don't do this. The idea to use low priority for the scans was to not starve user queries or point operations.